src/util/modules.py

```diff
@@ -721,14 +730,24 @@ def _force_install_module(cr, module, if_installed=None, reason="it has been exp
             [toinstall, list(INSTALLED_MODULE_STATES)],
         )
         for (mod,) in cr.fetchall():
-            _force_install_module(
-                cr,
-                mod,
-                reason=(
-                    "it is an auto install module that got all its auto install dependencies installed "
-                    "by the force install of {!r}"
-                ).format(module),
-            )
+            if mod in ENVIRON["__modules_to_skip_autoinstall"]:
+                _logger.info(
+                    (
+                        "Skipping the auto installation of module %r despite having all its auto install dependencies installed "
+                        "by the force install of %r, because it was previously uninstalled."
```

Comment:
We should make this more direct:
```suggestion
                        "Module %r won't be auto installed because its original dependencies were already installed. "
                        "Yet all its auto install dependencies were installed by the force install of {!r}."                        
```
* First we tell what we skip and why: no module installed
* Then we we tell how we reached this point: as a result of an install of another module.

Both details are highly useful when debugging auto-install issues.